File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/annotation/WrapMethod.java
package rbq.wtf.lycoris.client.wrapper.wrappers.annotation;

import rbq.wtf.lycoris.client.wrapper.MapEnum;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.repeat.WrapMethods;

import java.lang.annotation.*;

@Documented
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
@Repeatable(WrapMethods.class)
/**
 * Auto setup method
 */
public @interface WrapMethod {
    String mcpName();
    MapEnum targetMap();
    String customSrgName() default "none";
    String signature() default "none";
}

File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/annotation/WrapMethod.kt
package rbq.wtf.lycoris.client.wrapper.wrappers.annotation

import rbq.wtf.lycoris.client.wrapper.MapEnum
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.repeat.WrapMethods

@MustBeDocumented
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
@JvmRepeatable(
    WrapMethods::class
)
/**
 * Auto setup method
 */
annotation class WrapMethod(
    val mcpName: String,
    val targetMap: MapEnum,
    val customSrgName: String = "none",
    val signature: String = "none"
)